src/hera.cpp

```diff
@@ -130,15 +171,32 @@ static struct evm_result evm_execute(
     heraAssert(rev == EVM_BYZANTIUM, "Only Byzantium supported.");
 
     vector<uint8_t> _code(code, code + code_size);
+
+    if (msg->kind == EVM_CREATE) {
+      // Meter the deployment (constructor) code
+      _code = sentinel(context, _code);
+      heraAssert(_code.size() > 5, "Invalid contract or metering failed.");
+    }
+
     execute(context, _code, *msg, result);
 
     // copy call result
     if (result.returnValue.size() > 0) {
-      ret.output_size = result.returnValue.size();
+      vector<uint8_t> returnValue;
+
+      if (msg->kind == EVM_CREATE && !result.isRevert) {
```

Comment:
Does the `.isRevert`  also mean a "failure"?

Reply:
Yes. Real failures are caught by the exception block, here only two cases can happen success or revert.